src/simulation/messaging/messages.rs

```diff
@@ -422,6 +435,21 @@ impl Plan {
     }
 
     fn get_full_plan_for_routing(io_plan: &IOPlan, id_mappings: &MatsimIdMappings) -> Plan {
+        if io_plan.elements.is_empty() {
```

Comment:
One more reason to move the routing logic somewhere else. It is getting more and more messy.